prow/plugins/slackevents/slackevents.go

```diff
@@ -131,24 +133,46 @@ func notifyOnSlackIfManualMerge(pc client, pe github.PushEvent) error {
 }
 
 func isWhiteListed(mw *plugins.MergeWarning, pe github.PushEvent) bool {
-	bwl := mw.BranchWhiteList[pe.Branch()]
-	inWhiteList := stringInArray(pe.Pusher.Name, mw.WhiteList) || stringInArray(pe.Sender.Login, mw.WhiteList)
-	inBranchWhiteList := stringInArray(pe.Pusher.Name, bwl) || stringInArray(pe.Sender.Login, bwl)
+	wl := sets.String{}
+	for _, login := range mw.WhiteList {
+		wl.Insert(github.NormLogin(login))
+	}
+
+	bwl := sets.String{}
+	for _, login := range mw.BranchWhiteList[pe.Branch()] {
+		bwl.Insert(github.NormLogin(login))
+	}
+
+	inWhiteList := wl.HasAny(github.NormLogin(pe.Pusher.Name), github.NormLogin(pe.Sender.Login))
+	inBranchWhiteList := bwl.HasAny(github.NormLogin(pe.Pusher.Name), github.NormLogin(pe.Sender.Login))
 	return inWhiteList || inBranchWhiteList
 }
 
 func getMergeWarning(mergeWarnings []plugins.MergeWarning, org, repo string) *plugins.MergeWarning {
+	fullName := fmt.Sprintf("%s/%s", org, repo)
+
+	// First search for repo config
 	for _, mw := range mergeWarnings {
-		if stringInArray(org, mw.Repos) || stringInArray(fmt.Sprintf("%s/%s", org, repo), mw.Repos) {
-			return &mw
+		if !strInSlice(fullName, mw.Repos) {
+			continue
 		}
+		return &mw
 	}
+
+	// If you don't find anything, loop again looking for an org config
+	for _, mw := range mergeWarnings {
+		if !strInSlice(org, mw.Repos) {
+			continue
+		}
+		return &mw
+	}
+
 	return nil
 }
 
-func stringInArray(str string, list []string) bool {
-	for _, v := range list {
-		if v == str {
+func strInSlice(str string, slice []string) bool {
```

Comment:
aside: do we have a util for this somewhere? seems goofy to have our own method for this littered everywhere 🤷‍♂

Reply:
I'm not sure. Yeah, it's everywhere, and obv not consistently.

Reply:
✨ golang  ✨

Reply:
You're already importing `sets` -- why not `sets.NewString(mw.Repos).Has(fullName)` instead of calling this helper?